### PR TITLE
Use `Lazy<Cid>` for `GENESIS_CID`s rather than a string

### LINKED ISSUE #3230: refactor: `GENESIS_CID: Lazy<Cid>` rather than a string
https://github.com/ChainSafe/forest/blob/df5c5a38b9d4189654f5bb47c9129ebdc2c84712/src/networks/calibnet/mod.rs#L13-L14

Genesis CIDs for mainnet and calibnet are given as strings. This is quite awkward and `Lazy<Cid>` would be nicer. Of course, plain `Cid` would be best, but that's another can of worms.

_Originally posted by @aatifsyed in https://github.com/ChainSafe/forest/pull/3220#discussion_r1267384871_

### PR DESCRIPTION
**Pull request by jdjaustin**
## Summary of changes

Changes introduced in this pull request:

- Change `GENESIS_CID`s to use `Lazy<Cid>` type rather than a string slice.

## Reference issue to close (if applicable)

Closes #3230 

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/blocks/tipset.rs [hunk 1]
```diff
@@ -284,8 +284,8 @@ impl Tipset {
             serde_yaml::from_str(include_str!("../../build/known_blocks.yaml")).unwrap()
         });
 
-        let calibnet_cid = Cid::from_str(calibnet::GENESIS_CID).unwrap();
-        let mainnet_cid = Cid::from_str(mainnet::GENESIS_CID).unwrap();
+        let calibnet_cid = *Lazy::get(&calibnet::GENESIS_CID).expect("failed to get calibnet genesis CID");
```
Comment (lemmih) [inline src/blocks/tipset.rs:287]:
```suggestion
        let calibnet_cid = *calibnet::GENESIS_CID;
```

File: src/blocks/tipset.rs [hunk 2]
```diff
@@ -284,8 +283,8 @@ impl Tipset {
             serde_yaml::from_str(include_str!("../../build/known_blocks.yaml")).unwrap()
         });
 
-        let calibnet_cid = Cid::from_str(calibnet::GENESIS_CID).unwrap();
-        let mainnet_cid = Cid::from_str(mainnet::GENESIS_CID).unwrap();
+        let calibnet_cid = *calibnet::GENESIS_CID;
+        let mainnet_cid = *mainnet::GENESIS_CID;
```
Comment (lemmih) [inline src/blocks/tipset.rs:287]: Let's inline these two variables.

**Review by lemmih — APPROVED** [review]

File: src/networks/mod.rs [hunk 3]
```diff
@@ -176,7 +176,7 @@ impl ChainConfig {
         use mainnet::*;
         Self {
             network: NetworkChain::Mainnet,
-            genesis_cid: Some(GENESIS_CID.to_owned()),
```
Comment (LesnyRumcajs) [inline src/networks/mod.rs:179]: How about we go ahead and make the `genesis_cid` field in the `ChainConfig` an `Option<Cid>`? The more strongly typed the codebase is, the better.

**Review by LesnyRumcajs — APPROVED** [review]

File: src/networks/mod.rs (hunk 3 above)
Comment (LesnyRumcajs) [reply src/networks/mod.rs:179]: I'm okay with doing this in a separate PR to keep things small. Plus, the change may sprawl into several files.

Comment (jdjaustin) [reply src/networks/mod.rs:179]: Issue #3247

**Pull request merged**